internal/pkg/agent/cmd/inspect.go

```diff
@@ -135,13 +141,47 @@ func inspectConfig(ctx context.Context, cfgPath string, opts inspectConfigOpts,
 		return err
 	}
 
-	if !opts.variables {
+	if !opts.variables && !opts.includeMonitoring {
 		return printConfig(fullCfg, l, streams)
 	}
 	cfg, err := getConfigWithVariables(ctx, l, cfgPath, opts.variablesWait)
 	if err != nil {
 		return err
 	}
+
+	if opts.includeMonitoring {
+		// Load the requirements before trying to load the configuration. These should always load
+		// even if the configuration is wrong.
+		platform, err := component.LoadPlatformDetail()
+		if err != nil {
+			return fmt.Errorf("failed to gather system information: %w", err)
+		}
+		specs, err := component.LoadRuntimeSpecs(paths.Components(), platform)
+		if err != nil {
+			return fmt.Errorf("failed to detect inputs and outputs: %w", err)
+		}
```

Comment:
This is all duplicated below, should we just pull it above the if and share it?

Reply:
it would make it ugly, as one use case needs spec to proceed another does not, so if we share in between functions we would need to return monitoring cfg and spec

Reply:
Okay keep it how it is.